cmd/egress-cni-plugin/egressContext.go

```diff
@@ -233,101 +239,426 @@ func (c *EgressContext) SetupHostVeth(vethName string) error {
+		return err
+	}
+
+	ec.Log.Debugf("host IPv6 SNAT set up successfully")
+
+	mergeResult(ec.Result, ec.TmpResult)
+	// remove comment for debug purpose
+	// ec.Log.Debugf("output result: %+v", *ec.Result)
+
+	// Pass through the previous result
+	return types.PrintResult(ec.Result, ec.NetConf.CNIVersion)
+}
+
+// cmdDelEgressV6 exec clear the setting to support IPv6 egress traffic in EKS IPv4 cluster
+func (ec *egressContext) cmdDelEgressV6() (err error) {
+	var contIPAddrs []netlink.Addr
+
+	if ec.IPTablesIface == nil {
+		if ec.IPTablesIface, err = ec.IptCreator(iptables.ProtocolIPv6); err != nil {
+			ec.Log.Error("command ip6tables not found")
+		}
+	}
+
+	if ec.NsPath != "" {
+		err = ec.Ns.WithNetNSPath(ec.NsPath, func(hostNS ns.NetNS) error {
+			var containerLink netlink.Link
+			var _err error
+			ec.Log.Debugf("trying to get link by name %s", ec.NetConf.IfName)
+			containerLink, _err = ec.Link.LinkByName(ec.NetConf.IfName)
+			if _err != nil {
+				if !cniutils.IsLinkNotFoundError(_err) {
+					ec.Log.Debugf("failed to get link by name %s: %v", ec.NetConf.IfName, _err)
+				}
+				// must return when err, following code needs containerLink
+				return _err
 			}
 
-			if err != nil && err == ip.ErrLinkNotFound {
-				c.Log.Debugf("DEL: Link Not Found, returning", err)
-				return nil
+			contIPAddrs, _err = ec.Link.AddrList(containerLink, netlink.FAMILY_V6)
+			if _err != nil {
+				ec.Log.Debugf("failed to get container link %s IPv6 address, host may leave NAT rules uncleared: %v",
+					containerLink.Attrs().Name, _err)
+				return _err
 			}
-			return err
+
+			_err = ec.Link.LinkDel(containerLink)
+			if _err != nil {
+				if !cniutils.IsLinkNotFoundError(_err) {
+					ec.Log.Debugf("failed to delete veth %s in container: %v", containerLink.Attrs().Name, _err)
+				}
+			} else {
+				ec.Log.Debugf("Successfully deleted veth %s in container", containerLink.Attrs().Name)
+			}
+			return _err
 		})
 
-		//DEL should be best-effort. We should clean up as much as we can and avoid returning error
 		if err != nil {
-			c.Log.Debugf("DEL: Executing in container ns errored out, returning", err)
+			ec.Log.Errorf("failed to clean up link %s in container %s", ec.NetConf.IfName, ec.NsPath)
 		}
 	}
 
-	if c.NetConf.NodeIP != nil {
-		c.Log.Debugf("DEL: SNAT setup, let's clean them up. Size of ipnets: %d", len(ipnets))
-		for _, ipn := range ipnets {
-			if err := snat.Del(c.IpTablesIface, ipn.IP, c.SnatChain, c.SnatComment); err != nil {
+	err = ec.cleanupSnat(false, contIPAddrs)
+	if err != nil {
+		ec.Log.Errorf("Delete host SNAT chain/rule for ip %s failed: %v.", contIPAddrs, err)
+	}
+
+	return nil
+}
+
+func (ec *egressContext) disableInterfaceIPv6(ifName string) error {
+	return ec.Ns.WithNetNSPath(ec.NsPath, func(hostNS ns.NetNS) error {
+		var entry = "net/ipv6/conf/" + ifName + "/disable_ipv6"
+		if content, err := ec.Procsys.Get(entry); err == nil {
+			if strings.TrimSpace(content) == "1" {
+				return nil
+			}
+		}
```

Comment:
nit: since checking does a syscall as well, we may as well just blindly set it to 1 to decrease the number of syscalls

Reply:
This is a direct translation of logic [here](https://github.com/containernetworking/plugins/blob/main/pkg/ip/ipforward_linux.go#L56-L61), the translation is to make it unit testable.